### PR TITLE
Add report card lost or damaged

### PR DESCRIPTION
**Pull request by JKobrynski**
### Fixed Issues

$ #22879 
PROPOSAL: N/A

### Tests

1. Add the following code to **InitialSettingsPage.js**

```js
useEffect(() => {
    window.Onyx.merge(`cardList`, {
        virtual: {
            cardID: 234523452345,
            state: 3,
            bank: 'Expensify',
            availableSpend: 10000,
            domainName: 'Expensify',
            lastFourPAN: '',
            isVirtual: true,
            cardholderFirstName: 'Test',
            cardholderLastName: 'Test',
        },
        physical: {
            cardID: 234523452345,
            state: 3,
            bank: '1000',
            availableSpend: 10000,
            domainName: 'Expensify',
            lastFourPAN: '2345',
            isVirtual: false,
        },
    });

    Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute('expensify'));
}, []);
```

2. Open settings, you should be automatically navigated to the cards page
3. Click the "Report physical card loss / damage" button
4. You should be navigated to a new page that will allow you to select the reason
5. Click the "Next" button
6. You should see a button with your address, if you haven't added your address yet, you can press it to do so
7. On submit, it should load for a second and then navigate back. If there is no address, an error should be displayed.

(Check recordings, they explain a lot)

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/44364829/b433f179-52d1-4077-bc5d-4df215fd691d
https://github.com/Expensify/App/assets/44364829/7c409b2a-b378-487d-af12-0f952759d386

</details>

<details>
<summary>Mobile Web - Chrome</summary>

[chrome1-compressed.webm](https://github.com/Expensify/App/assets/44364829/2493713d-be57-445b-889f-87ea52b64ed2)
[chrome2-compressed.webm](https://github.com/Expensify/App/assets/44364829/c075e29c-f491-4491-a29e-1d903bcdaf72)

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/44364829/1807dd43-2d6c-40ae-a8d4-84c7703eda99
https://github.com/Expensify/App/assets/44364829/28805647-4615-4158-8f9e-214be8492f2c

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/44364829/619afb13-0444-407d-85cc-12fb5ae21ae2
https://github.com/Expensify/App/assets/44364829/2e8af130-ef20-47c7-a1a4-46dc4f5487c0

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/44364829/c7efda39-3dc0-49bd-a8c4-38309cc3706d
https://github.com/Expensify/App/assets/44364829/fc537680-e3ba-4faf-838e-780c68f3c80f

</details>

<details>
<summary>Android</summary>

[android1-compressed.webm](https://github.com/Expensify/App/assets/44364829/fc354b53-8ef7-41c1-96de-1eab12962f46)
[android2-compressed.webm](https://github.com/Expensify/App/assets/44364829/e5352685-7230-48ad-ba1f-44521657f7aa)

</details>

### CODE REVIEW

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (pasyukevich) [inline src/pages/settings/Wallet/ReportCardLostPage.js:142]: Missed `testID` on the wrapper

File: src/ONYXKEYS.ts
Comment (fabioh8010) [inline src/ONYXKEYS.ts:294]:
```suggestion
        REPORT_PHYSICAL_CARD_FORM: 'reportPhysicalCardLostForm',
```

File: src/components/SingleOptionSelector.js
Comment (fabioh8010) [inline src/components/SingleOptionSelector.js:60]: We can use the `useLocalize` hook instead of the this HOC.

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js:142]: Done, thanks!

File: src/components/Form.js
Comment (JKobrynski) [reply src/components/Form.js]: Right, it wasn't necessary, I found another way to achieve the same outcome

File: src/ONYXKEYS.ts
Comment (JKobrynski) [reply src/ONYXKEYS.ts]: Done 😄

File: src/ONYXKEYS.ts
Comment (MrMuzyk) [inline src/ONYXKEYS.ts:294]: `REPORT_PHYSICAL_CARD_LOST_FORM` ?

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (MrMuzyk) [inline src/pages/settings/Wallet/ReportCardLostPage.js:107]: this should probably be the first `if` to return as early as possible

**Comment by grgia** [comment]
Assigning @parasharrajat for C+ review

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js:107]: Done!

File: src/ONYXKEYS.ts
Comment (JKobrynski) [reply src/ONYXKEYS.ts]: Done!

File: src/components/SingleOptionSelector.js
Comment (JKobrynski) [reply src/components/SingleOptionSelector.js:60]: Unfortunately, it's not possible as of now, it causes errors. @rezkiy37 and I investigated it, and it's probably due to **Form.js** using `withLocalize` even though it's already a functional component.

**Comment by JKobrynski** [comment]
@parasharrajat @allroundexperts the PR is ready for review! I added screen recording for all platforms and completed the checklist

**Comment by JKobrynski** [comment]
@allroundexperts bump 😄

**Review by allroundexperts — CHANGES_REQUESTED** [review]
Some code refactors required.

File: src/languages/en.ts
Comment (JKobrynski) [reply src/languages/en.ts]: Removed!

File: src/languages/es.ts
Comment (JKobrynski) [reply src/languages/es.ts]: Removed!

File: src/libs/actions/Card.js
Comment (JKobrynski) [reply src/libs/actions/Card.js]: No reason at all, nice catch!

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js]: Done 😄

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js:139]: If you take a look at the `onSubmit` function, it passes the `reason` to `requestReplacementExpensifyCard` method, does that answer your question?

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js:135]: I split it into two separate methods depending on step, let me know what you think

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js]: I simplified it a bit and split into two smaller effects, let me know your thoughts 😄

File: src/libs/CardUtils.ts
Comment (JKobrynski) [reply src/libs/CardUtils.ts]: Yes, we have something similar in `PersonalDetailsInitialPage`. Take a look at [this PR](https://github.com/Expensify/App/pull/26778), where this method was modified and made reusable (and approved). I used the same solution for this PR, let me know what you think!

Comment (JKobrynski) [reply src/libs/CardUtils.ts]: I removed this method, so I guess it's outdated 😄

File: src/libs/CardUtils.ts
Comment (JKobrynski) [reply src/libs/CardUtils.ts]: Method removed (explained in other comments)

File: src/components/SingleOptionSelector.js
Comment (JKobrynski) [reply src/components/SingleOptionSelector.js:10]: Unfortunately it's not possible, it causes an error (discussed in [this comment](https://github.com/Expensify/App/pull/28141#discussion_r1340921737)). It's probably caused by `Form` that uses `withLocalize` and wraps `SingleOptionSelector` in that case.

**Comment by JKobrynski** [comment]
@allroundexperts PR updated!

**Comment by JKobrynski** [comment]
@allroundexperts lint issues fixed!

**Review by allroundexperts — CHANGES_REQUESTED** [review]
Changes needed!

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (JKobrynski) [reply src/pages/settings/Wallet/ReportCardLostPage.js]: Updated, however I had to do it that way:

```javascript
if (!_.isEmpty(physicalCard.errors) || !(prevIsLoading && !formData.isLoading)) {
    return;
}

Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
```

**Comment by JKobrynski** [comment]
@allroundexperts thanks for so many useful comments! I made many changes to address them, so here is a quick summary:

1. Simplified `useEffect` in `ReportCardLostPage`
2. Edited `Card.js` to reset error in optimistic data
3. As for the titles that might have wrong font size/family, I used Figma to check the details (they might be outdated) and it's `Expensify New Kansas`, 22px exactly like in the app
4. There is no default selected option in `SingleOptionSelector` (nothing is selected by default)
5. Added error handling for the reason step (no selected reason - show error)
6. Removed left margin from `SingleOptionSelector` (it was added by mistake, was never in designs)
7. Made only text + `SelectCircle` clickable in `SingleOptionSelector` instead of the whole row
8. Improved back button press handler, to allow users to go back to the previous step
9. Fixed the hover on address `MenuItem`
10. The console error must've been related to something else, I merged the latest main branch and it disappeared for me

Additionally:

I also worked on [add “Report virtual card fraud” page and route](https://github.com/Expensify/App/pull/28312) and I thought [this comment](https://github.com/Expensify/App/pull/28312#discussion_r1345031981) also applies to this PR, so I removed `Form` and used `FormAlertWithSubmitButton` directly, let me know what you think!

**Comment by shawnborton** [comment]
> Not sure if this margin left was in the designs. @shawnborton to confirm this..

Agree, no need for left margin.

So far the headline font style looks correct to me. The only minor feedback is that I think we want 12px of padding below the page header bar and above the headline.

For this screen here:
image


I think we want the description/paragraph text to appear above the push to page row. And I agree with your comments that it should stretch full edge-to-edge.

**Comment by shawnborton** [comment]
For this component here:
CleanShot 2023-10-06 at 07 58 34@2x


Why isn't this using our standard option row component? It should have bold text and slightly different alignment. Let's reuse existing components here if we can.

**Comment by JKobrynski** [comment]
@shawnborton added extra padding on top, used the existing component you asked for 😄

**Comment by shawnborton** [comment]
Awesome, are the screenshots updated too?

**Comment by JKobrynski** [comment]
@allroundexperts @shawnborton conflicts resolved, screen recordings updated!

**Comment by shawnborton** [comment]
Nice, this feels pretty good to me!

**Comment by JKobrynski** [comment]
@allroundexperts I updated text styles, it's now fixed 😄 (attaching a screenshot). About that transition, I don't think it was required (unless that changed) I even asked @grgia about it in [this thread](https://expensify.slack.com/archives/C027726KHT4/p1694519993227079), and she confirmed it can be one screen, let me know what you think  though!

Screenshot_1696923479

File: src/components/SingleOptionSelector.js
Comment (marcaaron) [reply src/components/SingleOptionSelector.js:10]: I think we can address this in a follow up and this is not a blocker for now.

File: src/ONYXKEYS.ts
Comment (marcaaron) [inline src/ONYXKEYS.ts:294]: This should be `requestPhysicalCardForm`

**Review by marcaaron — CHANGES_REQUESTED** [review]
Changes look good. Similar to https://github.com/Expensify/App/pull/28312 we will want to do internal QA on the changes.

File: src/ONYXKEYS.ts
Comment (JKobrynski) [reply src/ONYXKEYS.ts]: Done!

**Comment by JKobrynski** [comment]
@allroundexperts conflicts resolved

**Comment by joelbettner** [comment]
Just a heads up, it looks like the testing of this PR created this issue: https://github.com/Expensify/Expensify/issues/325564

`RequestReplacementExpensifyCard` was called without a `cardID`.

**Review by allroundexperts — APPROVED** [review]
Looks good!

File: src/components/SelectCircle.js
Comment (grgia) [inline src/components/SelectCircle.js:25]: Shouldn't props.styles be spread since it's an array?

**Review by grgia — COMMENTED** [review]
minor comment

File: src/components/SelectCircle.js
Comment (JKobrynski) [reply src/components/SelectCircle.js]: Done

**Review by grgia — DISMISSED** [review]

**Review by grgia — COMMENTED** [review]
image

image

Looks like the border color for the circle toggle is off- it should be our `icons` color

**Review by grgia — COMMENTED** [review]
@JKobrynski can you confirm this test

If reporting as lost/stolen, then physical card is removed from card page

image


If reporting as damaged, then physical card remains on card page

image

**Comment by JKobrynski** [comment]
@grgia border color changed! Regarding the second comment: yes, that's how I understand it based on the google doc, however we established that this PR won't implement full backend functionality (just boilerplate) as I am not able to test this on my end, and I don't even know if that API method is implemented yet, hope that answers your question!

**Review by grgia — APPROVED** [review]

**Comment by grgia** [comment]
Will let @marcaaron take a look before merging so I don't dismiss his review

**Review by marcaaron — APPROVED** [review]

File: src/pages/settings/Wallet/ReportCardLostPage.js
Comment (ishpaul777) [inline src/pages/settings/Wallet/ReportCardLostPage.js:155]: {BZ CHECKLIST} coming from https://github.com/Expensify/App/issues/63222, this should navigate back to SETTINGS_WALLET_DOMAIN_CARD route we fixed it in this [PR](https://github.com/Expensify/App/pull/63754)

**Pull request merged**
